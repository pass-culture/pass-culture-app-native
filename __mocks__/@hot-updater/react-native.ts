import { ReactNode } from 'react'

const getUpdateSource = jest.fn(() => Promise.resolve({ source: 'source' }))

const HotUpdater = jest.fn(() => (app: ReactNode) => app)

export default Object.assign(HotUpdater, {
  getUpdateSource,
})
