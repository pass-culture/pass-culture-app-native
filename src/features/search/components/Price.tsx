import { t } from '@lingui/macro'
import React from 'react'

import { _ } from 'libs/i18n'
import { Slider } from 'ui/components/inputs/Slider'

import { CenteredSection } from '../atoms/Sections'

const MAX_PRICE = 300
const formatEuro = (price: number) => `${price} â‚¬`

export const Price: React.FC = () => (
  <CenteredSection title={_(t`Prix`)}>
    <Slider showValues={true} values={[0, MAX_PRICE]} max={MAX_PRICE} formatValues={formatEuro} />
  </CenteredSection>
)
