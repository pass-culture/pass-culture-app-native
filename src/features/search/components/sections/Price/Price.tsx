import React, { useCallback } from 'react'

import { FilterRow } from 'features/search/components/FilterRow/FilterRow'
import { useSearch } from 'features/search/context/SearchWrapper'
import { FilterBehaviour } from 'features/search/enums'
import { getPriceAsNumber } from 'features/search/helpers/getPriceAsNumber/getPriceAsNumber'
import { getPriceDescription } from 'features/search/helpers/getPriceDescription/getPriceDescription'
import { PriceModal } from 'features/search/pages/modals/PriceModal/PriceModal'
import { useModal } from 'ui/components/modals/useModal'
import { Code } from 'ui/svg/icons/Code'

type Props = {
  onClose?: VoidFunction
}

export const Price = ({ onClose }: Props) => {
  const { searchState } = useSearch()
  const {
    visible: searchPriceModalVisible,
    showModal: showSearchPriceModal,
    hideModal: hideSearchPriceModal,
  } = useModal(false)

  const minPrice: number | undefined = getPriceAsNumber(searchState.minPrice)
  const maxPrice: number | undefined = getPriceAsNumber(searchState.maxPrice)

  const onPress = useCallback(() => {
    showSearchPriceModal()
  }, [showSearchPriceModal])

  return (
    <React.Fragment>
      <FilterRow
        icon={Code}
        title="Prix"
        description={getPriceDescription(minPrice, maxPrice)}
        onPress={onPress}
      />
      <PriceModal
        title="Prix"
        accessibilityLabel="Ne pas filtrer sur les prix et retourner aux rÃ©sultats"
        isVisible={searchPriceModalVisible}
        hideModal={hideSearchPriceModal}
        filterBehaviour={FilterBehaviour.APPLY_WITHOUT_SEARCHING}
        onClose={onClose}
      />
    </React.Fragment>
  )
}
