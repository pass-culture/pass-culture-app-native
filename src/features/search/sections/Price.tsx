import React from 'react'

import { CenteredSection, TitleWithCount } from 'features/search/atoms'
import { useStagedSearch } from 'features/search/pages/SearchWrapper'
import { SectionTitle } from 'features/search/sections/titles'
import { useLogFilterOnce } from 'features/search/utils/useLogFilterOnce'
import { useMaxPrice } from 'features/search/utils/useMaxPrice'
import { formatPriceInEuroToDisplayPrice } from 'libs/parsers'
import { Range } from 'libs/typesUtils/typeHelpers'
import { Slider } from 'ui/components/inputs/Slider'

export const Price: React.FC = () => {
  const logUseFilter = useLogFilterOnce(SectionTitle.Price)
  const { searchState, dispatch } = useStagedSearch()

  const maxPrice = useMaxPrice()
  const priceRange = searchState.priceRange ?? [0, maxPrice]
  const count = +(priceRange[0] > 0 || priceRange[1] < maxPrice)
  const values = searchState.offerIsFree ? [0, 0] : priceRange

  const onValuesChangeFinish = (newValues: number[]) => {
    const priceRangeIsFree = newValues[0] === 0 && newValues[1] === 0
    if (priceRangeIsFree && !searchState.offerIsFree) dispatch({ type: 'TOGGLE_OFFER_FREE' })
    if (!priceRangeIsFree && searchState.offerIsFree) dispatch({ type: 'TOGGLE_OFFER_FREE' })
    dispatch({ type: 'PRICE_RANGE', payload: newValues as Range<number> })
    logUseFilter()
  }

  return (
    <CenteredSection title={<TitleWithCount title={SectionTitle.Price} count={count} />}>
      <Slider
        showValues={true}
        values={values}
        max={maxPrice}
        formatValues={formatPriceInEuroToDisplayPrice}
        onValuesChangeFinish={onValuesChangeFinish}
      />
    </CenteredSection>
  )
}
