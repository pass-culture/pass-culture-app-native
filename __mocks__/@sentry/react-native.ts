import * as ActualSentryModule from '@sentry/react-native'

const MockedSentryModule = jest.genMockFromModule<typeof ActualSentryModule>('@sentry/react-native')

module.exports = {
  ...MockedSentryModule,
}
