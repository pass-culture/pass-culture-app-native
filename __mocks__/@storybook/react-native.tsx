import React from 'react'; // @storybook
import { View } from 'react-native';

const configure = jest.fn();
const getStorybookUI = jest.fn().mockImplementation(() => () => <View />);

export { configure, getStorybookUI };
