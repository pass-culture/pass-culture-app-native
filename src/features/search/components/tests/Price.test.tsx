import { render } from '@testing-library/react-native'
import React from 'react'

import { Price } from '../Price'

describe('Price component', () => {
  it('should render correctly', () => {
    const { toJSON, queryByText } = render(<Price />)
    expect(toJSON()).toMatchSnapshot()
    expect(queryByText('0 € - 300 €')).toBeTruthy()
  })
})
