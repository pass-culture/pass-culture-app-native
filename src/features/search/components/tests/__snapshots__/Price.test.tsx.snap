// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price component should render correctly 1`] = `
Array [
  <Text
    style={
      Array [
        Object {
          "color": "#151515",
          "fontFamily": "Montserrat-Medium",
          "fontSize": 18,
          "lineHeight": 22,
        },
      ]
    }
  >
    Prix
  </Text>,
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "marginTop": 16,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#151515",
            "fontFamily": "Montserrat-Bold",
            "fontSize": 15,
            "lineHeight": 20,
          },
        ]
      }
    >
      0 € - 300 €
    </Text>
    <View
      numberOfSpaces={4}
      style={
        Array [
          Object {
            "height": 16,
          },
        ]
      }
    />
    <View>
      <View
        style={
          Array [
            Object {
              "height": 50,
              "justifyContent": "center",
              "position": "relative",
            },
            Object {
              "height": 20,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              Object {
                "width": 280,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#A7A7A7",
                  "borderRadius": 2,
                  "height": 2,
                },
                Object {
                  "borderRadius": 24,
                  "height": 15,
                  "marginTop": -7,
                },
                Object {
                  "backgroundColor": "#c7c7cc",
                },
                Object {
                  "width": 0,
                },
              ]
            }
          />
          <View
            onMoveShouldSetResponder={[Function]}
            onMoveShouldSetResponderCapture={[Function]}
            onResponderEnd={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderReject={[Function]}
            onResponderRelease={[Function]}
            onResponderStart={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onStartShouldSetResponderCapture={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "#A7A7A7",
                  "borderRadius": 2,
                  "height": 2,
                },
                Object {
                  "borderRadius": 24,
                  "height": 15,
                  "marginTop": -7,
                },
                Object {
                  "backgroundColor": "#eb0055",
                },
                Object {
                  "width": 280,
                },
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#A7A7A7",
                  "borderRadius": 2,
                  "height": 2,
                },
                Object {
                  "borderRadius": 24,
                  "height": 15,
                  "marginTop": -7,
                },
                Object {
                  "backgroundColor": "#c7c7cc",
                },
                Object {
                  "width": 0,
                },
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "height": 48,
                  "justifyContent": "center",
                  "position": "absolute",
                  "width": 48,
                },
                Object {
                  "left": -24,
                  "top": -24,
                },
                undefined,
                false,
              ]
            }
          >
            <View
              onMoveShouldSetResponder={[Function]}
              onMoveShouldSetResponderCapture={[Function]}
              onResponderEnd={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderReject={[Function]}
              onResponderRelease={[Function]}
              onResponderStart={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              onStartShouldSetResponderCapture={[Function]}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "stretch",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  },
                  Object {
                    "borderRadius": 15,
                  },
                ]
              }
            >
              <View
                accessible={true}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
              >
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#FFFFFF",
                        "borderColor": "#DDDDDD",
                        "borderRadius": 30,
                        "borderWidth": 1,
                        "height": 30,
                        "shadowColor": "#000000",
                        "shadowOffset": Object {
                          "height": 3,
                          "width": 0,
                        },
                        "shadowOpacity": 0.2,
                        "shadowRadius": 1,
                        "width": 30,
                      },
                      Object {
                        "backgroundColor": "#ffffff",
                        "borderColor": "#ffffff",
                        "borderRadius": 28,
                        "elevation": 4,
                        "height": 28,
                        "shadowColor": "#151515",
                        "shadowOpacity": 0.2,
                        "shadowRadius": 4,
                        "width": 28,
                      },
                      undefined,
                      undefined,
                    ]
                  }
                />
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "height": 48,
                  "justifyContent": "center",
                  "position": "absolute",
                  "width": 48,
                },
                Object {
                  "right": -24,
                  "top": -24,
                },
                undefined,
              ]
            }
          >
            <View
              onMoveShouldSetResponder={[Function]}
              onMoveShouldSetResponderCapture={[Function]}
              onResponderEnd={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderReject={[Function]}
              onResponderRelease={[Function]}
              onResponderStart={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              onStartShouldSetResponderCapture={[Function]}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "stretch",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  },
                  Object {
                    "borderRadius": 15,
                  },
                ]
              }
            >
              <View
                accessible={true}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
              >
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#FFFFFF",
                        "borderColor": "#DDDDDD",
                        "borderRadius": 30,
                        "borderWidth": 1,
                        "height": 30,
                        "shadowColor": "#000000",
                        "shadowOffset": Object {
                          "height": 3,
                          "width": 0,
                        },
                        "shadowOpacity": 0.2,
                        "shadowRadius": 1,
                        "width": 30,
                      },
                      Object {
                        "backgroundColor": "#ffffff",
                        "borderColor": "#ffffff",
                        "borderRadius": 28,
                        "elevation": 4,
                        "height": 28,
                        "shadowColor": "#151515",
                        "shadowOpacity": 0.2,
                        "shadowRadius": 4,
                        "width": 28,
                      },
                      undefined,
                      undefined,
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>,
]
`;
